service/history/queues/dlq_writer.go

```diff
@@ -99,5 +109,26 @@ func (q *DLQWriter) WriteTaskToDLQ(ctx context.Context, sourceCluster, targetClu
 	if err != nil {
 		return fmt.Errorf("%w: %v", ErrSendTaskToDLQ, err)
 	}
+	q.metricsHandler.Counter(metrics.DLQWrites.GetMetricName()).Record(1)
+	ns, err := q.namespaceRegistry.GetNamespaceByID(namespace.ID(task.GetNamespaceID()))
+	if err != nil {
+		q.logger.Warn("Failed to get namespace name while trying to write a task to DLQ",
+			tag.WorkflowNamespace(task.GetNamespaceID()),
+			tag.Error(err),
+		)
+		q.logger.Warn("Task enqueued to DLQ",
+			tag.SourceCluster(sourceCluster),
+			tag.TargetCluster(targetCluster),
+			tag.TaskType(task.GetType()),
+			tag.WorkflowNamespaceID(task.GetNamespaceID()),
+		)
```

Comment:
I don't think we should duplicate this. Instead, just get a variable for the namespace tag, and set it to the namespace ID if we fail, or the namespace name if we succeed, then we don't need the early return here or the duplicated logger statement/tags.